### ISSUE TITLE
Alerts: implement server logic for alerting on block ranges

### ISSUE THREAD

**Issue by FreddieAkeroyd** [ISISComputingGroup/IBEX#3949]
This is required for LOQ, they have SECI alerts on Vacuum, Chopper etc. blocks but these are generally fixed ranges. SECI had a GUI for adding alerts, but for the moment we can just implement the server side of #3811 

The logic is pretty much run control logic, but with a different action - eventually a generic "block, condition, action" system would cover everything with action being "alert", "stop collection" or something else.

It may be possible to template out and use the existing run control logic, this should be investigated. Sending an alert can be done by using the webget module already built into our system, this module is currently used by isisdae to trigger a reset of the archiver via its web page.

Acceptance Criteria

* A block and alert ranges can be defined via PVs
* A who to alert PV exists for email and text messages

**Comment by kjwoodsISIS** [comment]
@FreddieAkeroyd - is the rework on this ticket complete?  Is it ready for a final review & closure?

**Comment by FreddieAkeroyd** [comment]
In progress - see  http://epics-jenkins.isis.rl.ac.uk/job/FreddieSystemTests/

**Comment by FreddieAkeroyd** [comment]
Above tests now complete

**Comment by DominicOram** [comment]
It's really hard to follow what's happening in this ticket and in #4000 as the PRs have all been prematurely merged and there is no link to the corresponding changes which have been made in the IOC submodule. However, the following is a review of the whole package of things:

The following code is in the st.cmd of the IOC and was presumably added as part of this ticket:
```
#dbLoadRecords("$(WEBGET)/db/sendAlert.db","P=$(MYPVPREFIX),Q=CS:DC:ALERTS:")
#dbLoadRecords("$(RUNCONTROL)/db/gencontrolMgr.db","P=$(MYPVPREFIX),MODE=DC,OUT_ACTION=$(MYPVPREFIX)CS:DC:ALERTS:MESSAGE:SP.PROC")
##dbLoadRecords("$(RUNCONTROL)/db/gencontrol.db","P=$(MYPVPREFIX),MODE=DC,PV=$(MYPVPREFIX)DAE:AD1:INTG:RATE")
#dbLoadRecords("$(RUNCONTROL)/db/gencontrol.db","P=$(MYPVPREFIX),MODE=DC,PV=$(MYPVPREFIX)CS:SB:GoodFrames")
```
What is it doing and why is it commented out?

The logic inside `LOQ_detector.db` is presumably only needed on LOQ. I can see that it's been hotfixed to be loaded on LOQ but given that the hotfix is not listed [here](https://github.com/ISISComputingGroup/IBEX/wiki#instrument-information--hotfixes) it's unlikely we'll remember to hotfix it again. Also, hotfixing it every cycle seems bad, what's the long-term plan for it to be loaded on LOQ but not elsewhere?

If I received a support call about this I don't think the documentation is sufficient for me to tell how to fix it, some immediate points:
* I normally look for documentation in the developers wiki rather than the main IBEX one. The developer's wiki still has no reference to how alerts work.
* There is a reference to `http://control-svcs.isis.cclrc.ac.uk/isiscgi/sendmess.py` where is this code in version control? What am I expected to pass to it? How do I know if it breaks? How can I deploy a new version if I need to?
* How does the code here, in #4000 and in the python script above link together to send an alert? A diagram would be useful here.
* It's not immediately obvious what ACTION_IN, ACTION_OUT and MODE do, can you add some documentation for these at the top of relevant DB?
* The documentation at https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Run-control should probably be updated to mention how the system is now generic and also works for alarms

**Comment by FreddieAkeroyd** [comment]
The commented out code is part of a test/example and can be removed, the code for alerting on blocks is already loaded separately and can be configured as per https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Email-Text-Alerts-from-IOCS though a diagram of how it all worked would be an improvement. It looks like I originally added this to the ibex rather than developers wiki, this was by accident and i have copied it across to the developers wiki. On LOQ they want to always alert on the count rate and also put in the aperture, that is done by loading the code as is on the instrument which is based on the commented out code above. A better solution than hotfixing does indeed need to be found, which includes deciding where such instrument files permanently live and are loaded. They could be in "settings" on an instrument or they could be with the ioc, both have advantages and disadvantages when upgrades and visibility etc. are considered. Also the logic of  loading might be better with a lua ioc for example. At the moment LOQ is the only place using alerts and just on the detector pv (i.e. no blocks), anybody else making use of them is waiting on a GUI mechanism being added.

**Comment by kjwoodsISIS** [comment]
What is stopping this ticket from progressing?

**Comment by kjwoodsISIS** [comment]
This ticket appears to have made no progress since 16th July.  What is preventing it from making progress?

**Comment by FreddieAkeroyd** [comment]
The LOQ detector code is now conditionally loaded with #480

**Comment by FreddieAkeroyd** [comment]
control-svcs sendmess.py docs at https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Alert-Relay

**Comment by FreddieAkeroyd** [comment]
Run control wiki page and db files comments updated
